config/php.md: new PHP docs to help users with the `php` metapackage transition (#731)
void-linux/void-docs: src/config/php.md

# PHP

There are two ways to install PHP packages with XBPS:

> 1. Using the versioned (otherwise known as pinned) packages (reccomended).

Reviewer: ```suggestion
1. Using the versioned (otherwise known as pinned) packages (recommended).
```

2. Using the metapackages.

> ## Pinned PHP Packages

Reviewer: ```suggestion
## Versioned/Pinned PHP Packages
```

Reviewer 2: I like Versioned PHP packages more, pinned implies something like holding or repolocking it in the package manager to me.

Reviewer: To be honest, that's also what I thought when I saw "pinned" being used here. It just seems out of place.

It is generally recommended to use versioned php packages (e.g. `php8.1`,
`php8.1-apcu`, etc.) for most use cases as this ensures a consistent environment
on updates with minimal or no intervention required.

> ## PHP Metapackages

Reviewer: ```suggestion
## PHP Meta-packages
```

In Void, the `php` package is a meta-package that points to the latest upstream
PHP version. This convention is followed by all packages prefixed with `php-`,
such as `php-fpm`, as well as `xdebug` and `composer`. See [the php
template](https://github.com/void-linux/void-packages/blob/master/srcpkgs/php/template)
for a complete list. It is recommended to only use these meta-packages for
development purposes.

When using a PHP meta-package, be warned that updating may require manual

> intervention if a new major PHP version has been included in the repository. As

Reviewer: Change "included" to "added" maybe?

a part of the version change, the configuration location will change to reflect
the new version. For example, upgrading from 8.0 to 8.1 would result in the

> config path changing from `/etc/php8.0` to `/etc/php8.1`. Any configurations
> that have been customized need to be manually applied to the new configuration
> directory.

Reviewer: I think this reads a little better, but feel free to ignore it: 
```suggestion
configuration path changing from `/etc/php8.0` to `/etc/php8.1`. Any
customizations that have been made need to be manually applied to the new
configuration directory.
```
